### PR TITLE
Separate system-probe config from datadog config

### PR DESCRIPTION
**Pull request by brycekahle**
### What does this PR do?

Reintroduces the changes from #13895 and #13994, along with a few fixes:

- Ignores failures to find the `system-probe.yaml` config file, even when a specific path is given. This is what was causing the kitchen test to fail, and the agent benchmarks to increase CPU by 20% (process-agent was in a crash loop).
- Fixes usage of the deprecated `--config` `process-agent` argument to use `--cfgpath` instead.
- Removes references to deprecated `process_agent_enabled` config key.
- Fixes feature detection to not run for `system-probe` config.
- Loads `datadog.yaml` config in `system-probe`, but will not fail if not present. This is for CWS specifically, since they have some subtle requirements on `datadog.yaml` values.

### Motivation

#13895 and #13994 were reverted in #14001 due to a failing kitchen test for process-agent startup.

### Describe how to test/QA your changes

A full CI pipeline has been started for this PR.

(Processes) Start the agent in docker, linux, and windows and make sure there is no warning regarding using the deprecated --config flag.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by paulcacheux — APPROVED** [review]

**Review by paulcacheux — COMMENTED** [review]
Actually there is an issue with this. In the CWS side we use the `hostname.Get` (see https://github.com/DataDog/datadog-agent/blob/4623adc3c6195a8c926df2ef70955cfeba7a68e7/pkg/security/probe/activity_dump_manager.go#L236) and in our tests it now fails saying "Trying to access features before detection has run". I think the core agent config is not correctly done

**Review by paulcacheux — CHANGES_REQUESTED** [review]
It fails in https://gitlab.ddbuild.io/DataDog/datadog-agent/-/jobs/185627265 for example

**Comment by brycekahle** [comment]
> I think the core agent config is not correctly done

system-probe shouldn't be reading `datadog.yaml` at all, which was part of the point of this PR. I see this is mostly for tags. Does this data flow through the security-agent and could the tags be read and applied there?

**Review by djmitche — APPROVED** [review]
Looks good to me.  Looks very familiar, so hopefully I didn't miss any subtle changes!

**Comment by paulcacheux** [comment]
I'm getting more and more nervous about this change, looking at CWS code I'm finding more and more places where we actually use the core config:
- hostname fetching in activity dumps
- remote configuration (creating a new client calls `GetDDAgentSecureClient` that fetches `config.Datadog.GetString("cmd_port")` and `Datadog.GetString("ipc_address")`.
Actually even the good way of fetching the hostname would involve the `GetDDAgentClient` that reads the core config as well (this is what is done by the process agent for example)

**Comment by djmitche** [comment]
Hm, so to summarize: conceptually security-agent doesn't access the core config, but it turns out that's not the case in a bunch of ways.

Is that a permanent state, or is it a long-term goal that all of those exceptions be eliminated?

Is there a way to model the current state?  And if it's a goal to eliminate exceptions, could we implement this in such a way that we don't accidentally introduce new exceptions?

**Comment by brycekahle** [comment]
I've bumped this to 7.42, so we have some time to figure out how to approach this.

File: comp/core/sysprobeconfig/component.go [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+	// These have had the EnvPrefix applied, as well as the EnvKeyReplacer.
+	GetEnvVars() []string
+
+	// IsSectionSet checks if a given section is set by checking if any of
+	// its sub-keys is set.
+	IsSectionSet(section string) bool
+
+	// Warnings returns config warnings collected during setup.
+	Warnings() *config.Warnings
+}
```
Comment (ogaca-dd) [inline comp/core/sysprobeconfig/component.go:114]: This interface seems to be the same as `comp/core/config/component.go`. You can use  `type Component config.Component` if they must be the same or the following code if this config must have all the methods from config.Component and other ones. 
``
type Component interface {
	config.Component
}
``
Note: See my comment comp/core/sysprobeconfig/config.go before applying a change.

File: comp/core/sysprobeconfig/config.go [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+		deps.Params.SysProbeConfFilePath,
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	return &cfg{warnings}, nil
+}
+
+func (c *cfg) IsSet(key string) bool {
```
Comment (ogaca-dd) [inline comp/core/sysprobeconfig/config.go:46]: The code is similar to comp/core/config/config.go. A common implementation should be used for both.

**Comment by brycekahle** [comment]
This has been rebased, and I've added the optional reading of `datadog.yaml` in system-probe as a fix until we can figure out how to restructure things for CWS.

**Review by xornivore — APPROVED** [review]
🙌 @brycekahle changes look great for Processes' owned files, just want to be safe since the changes are indeed pretty massive, anything I can help with testing this?

**Comment by brycekahle** [comment]
> anything I can help with testing this?

@xornivore I've run a full pipeline on it. I've also sanity checked that the `agent`, `process-agent`, and `system-probe` startup without error and can talk to each other. Any other ideas of things I should check?

**Comment by brycekahle** [comment]
Just to be clear, there is *no rush* to get this into 7.43. I'd rather be safe.

**Review by iglendd — APPROVED** [review]
LGTM

**Review by safchain — APPROVED** [review]

**Review by ogaca-dd — APPROVED** [review]

**Review by mfpierre — APPROVED** [review]

**Review by KSerrania — APPROVED** [review]

**Review by ahmed-mez — APPROVED** [review]

**Review by paulcacheux — APPROVED** [review]
LGTM thanks !

**Review by hmahmood — APPROVED** [review]

File: Dockerfiles/agent/s6-services/process/run [hunk 3]
```diff
@@ -1,4 +1,4 @@
 #!/usr/bin/execlineb -P
 
 foreground { /initlog.sh "starting process-agent" }
-process-agent --config=/etc/datadog-agent/datadog.yaml
+process-agent --cfgpath=/etc/datadog-agent/datadog.yaml
```
Comment (just-chillin) [inline Dockerfiles/agent/s6-services/process/run:4]: Thanks!!!

**Review by just-chillin — APPROVED** [review]
Looks good from the process side! Thanks so much for pushing this PR through

**Review by vboulineau — APPROVED** [review]

**Review by ISauve — APPROVED** [review]

**Review by carlosroman — APPROVED** [review]

**Pull request merged**